examples/barotropic_tracer_advection.jl

```diff
@@ -0,0 +1,22 @@
+using Oceananigans
+
+include("utils.jl")
+
+Nx, Ny, Nz = 128, 128, 1
+Lx, Ly, Lz = 2000, 2000, 10
+Nt, Δt = 25, 1
+
+model = Model(N=(Nx, Ny, Nz), L=(Lx, Ly, Lz), ν=4e-2, κ=4e-2)
+
+@. model.tracers.T.data[70:90, 10:30, 1] += 0.01
```

Comment:
Should we get into the habit of setting fields with functions? It seems potentially more robust. for example, check out the [`set!` function in `OceanTurb.jl`](https://github.com/glwagner/OceanTurb.jl/blob/35ce00da3ef1bdd707a618bce6c7e61fe994d6c5/src/fields.jl#L132), which combined with [an overloading of `setproperty!`](https://github.com/glwagner/OceanTurb.jl/blob/35ce00da3ef1bdd707a618bce6c7e61fe994d6c5/src/OceanTurb.jl#L127) enables syntax like

```
T0(x, y, z) = 285 + 0.01*exp( -(x^2+y^2+z^2) )
model.solution.T = T0
```
For example.

Reply:
Yes that would be awesome, then we can set using functions, constants, arrays, profiles, etc.